test/bdd/agent_steps.go

```diff
@@ -85,10 +85,13 @@ func (a *AgentSteps) registerPostMsgEvent(agentID, statesValue string) error {
 	states := strings.Split(statesValue, ",")
 	a.initializeStates(agentID, states)
 	go func() {
```

Comment:
(from a previous PR) I think it is generally cleaner to create a func and call `go funcname()`, rather than embed like this.

Reply:
yeah, I'll update this existing code.